packages/type-safe-api/src/construct/integrations/integration-response-set.ts

```diff
@@ -0,0 +1,187 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import {
+  ApiGatewayIntegrationResponse,
+  IntegrationRenderProps,
+} from "./integration";
+import { generateCorsResponseParameters } from "../prepare-spec-event-handler/prepare-spec";
+
+/**
+ * A set of integration responses. You can extend this to implement your own integration response set
+ */
+export abstract class IntegrationResponseSet {
```

Comment:
Given this doesn't implement anything, wouldn't it be better to model this as an interface?

Reply:
Probably should be an interface, though I quite like that this is consistent with the `Integration` abstract class (which is an abstract class because it has a default implementation of a `grant` method), and allows us to implement additional functionality into the class in the future without a breaking change. Though I don't foresee a need to add additional functionality any time soon...